### PR TITLE
chisel 1.7.1

### PR DESCRIPTION
**Pull request by kastiglione**
Update chisel for the [1.7.1 release](https://github.com/facebook/chisel/releases/tag/1.7.1).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by kastiglione** [comment]
@keith

**Comment by kastiglione** [comment]
Sorry for the thrash on this. I don't have an El Capitan machine to test with.

**Comment by ilovezfs** [comment]
code signed? How is anything being code signed here?

**Comment by kastiglione** [comment]
@ilovezfs By default, Xcode code signs simulator binaries. Chisel now includes a library that aids debugging, and that library is loaded dynamically into the app being debugged. This library is built by Xcode and so it too is code signed. It's not evident in this recipe, but the `make` command is calling `xcodebuild`, and it's `xcodebuild` that does the signing.

**Comment by kastiglione** [comment]
I will update the comment in the recipe.

**Comment by ilovezfs** [comment]
There are no signing certs installed on Jenkins so I'm still not clear what we're talking about.

**Comment by kastiglione** [comment]
Simulator builds use ad-hoc code signing, no signing identity is required.

**Comment by ilovezfs** [comment]
If this is running `make` it's no longer `bottle :unneeded`

**Comment by kastiglione** [comment]
Sounds good. How can I verify locally that a bottled install of this Chisel update works? I'd like to do that before this gets committed.

File: Formula/chisel.rb [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-  bottle :unneeded
-
   def install
     libexec.install Dir["*.py", "commands"]
     prefix.install "PATENTS"
+
+    # == LD_DYLIB_INSTALL_NAME Explanation ==
+    # This make invocation calls xcodebuild, which in turn performs ad hoc code
+    # signing. Note that ad hoc code signing does not need signing identities.
+    # Brew will update binaries to ensure their internal paths are usable, but
```
Comment (vinyldarkscratch) [inline Formula/chisel.rb]: The brew test bot isn't liking this one particular space at the very end!  Can you remove it real quick?

**Review by queengooborg — CHANGES_REQUESTED** [review]

File: Formula/chisel.rb (hunk 1 above)
Comment (kastiglione) [reply Formula/chisel.rb]: fixed, thanks.

Comment (vinyldarkscratch) [reply Formula/chisel.rb]: Perfect!  We'll let the brew test bot do it's thing now.  :wink:

**Review by queengooborg — APPROVED** [review]
CI checks are a go, so it's all good with me!

**Comment by kastiglione** [comment]
thanks @vinyldarkscratch. Do you know if there's a way I can install the bottle built by CI? I want to do a dry run. I don't see anything in `brew install` to specify a bottle location.

**Comment by vinyldarkscratch** [comment]
By default, Homebrew will install the bottle if it can find one in the formula (you can specify otherwise with the `--build-from-source` option).  Since this PR hasn't been merged yet, the test bot hasn't added the bottles, and as such there are none just yet!

**Comment by kastiglione** [comment]
ok thanks. I downloaded the bottle from CI, and `codesign --verify Chisel` results in:

> Chisel: invalid signature (code or signature have been modified)

I don't know what the cause is.

Is it an option to have this recipe be built by users and avoid bottling? The library is small, the build is quick.

Alternatively, the best solution may be to manually call `codesign` in `post_install`, to ensure it's always got a valid signature.

**Comment by vinyldarkscratch** [comment]
@kastiglione There is a way, yes, as described in the [Formula class documentation](http://www.rubydoc.info/github/Homebrew/brew/master/Formula#bottle-class_method).  You can add `bottle :disable, "reasons"` to your formula (which I noticed you had at one point, which leads me to the following part).  However, `formulae where the upstream URL breaks or moves frequently, require compile or have a reasonable amount of patches/resources should be bottled.`  Since `chisel` requires compilation, it should be bottled.

> How can I verify locally that a bottled install of this Chisel update works?

If you `pour chisel` with the edits on your local copy, does everything function the way you'd expect?  If so, then the bottled version should work perfectly.  Essentially, the bottle is grabbing the pre-compiled version of the library and all of it's files, and pouring them into a bottle to ship off to the computers.  (Couldn't help but keep up the beer analogy a bit.  😛)

For the codesigning, honestly I can't really say anything in regards to it.  I can only quote @ilovezfs in that "there are no signing certs installed on Jenkins," so honestly it seems like codesigning doesn't even happen aside from what the compiler does?



Bottom line, if an install of Chisel on your local machine seems to work as you expect, then I think this PR is ready to be pulled.  (That's when we can test to make sure the bottle functions, admittedly.)  Of course, that's just my opinion!

**Comment by kastiglione** [comment]
> If you `pour chisel` with the edits on your local copy, does everything function the way you'd expect?

When I run `brew install chisel` with these changes, it does work. (aside: I don't know what `pour chisel` is)

> That's when we can test to make sure the bottle functions, admittedly.

My initial test (using the bottle built by CI) indicates the bottle won't work, but maybe my test was flawed. Since the new library adds new features, users will generally not be impacted. Also, I am ready with next actions if the bottle does prove to be invalid, so let's go for it.

**Comment by vinyldarkscratch** [comment]
> `pour chisel`

Eek!  I'm so used to my aliases for different commands (`pour = brew install`), I keep forgetting that it's not what everyone uses!  My bad... 😅

And, yeah, I think that the best next step is to have this PR pulled, and then make any changes from there as needed.  :wink:

**Comment by kastiglione** [comment]
@ilovezfs Does this look acceptable?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for the upgrade @kastiglione! Shipped.

**Comment by kastiglione** [comment]
Thanks for the reviews and guidance @ilovezfs
